src/Transport/SnmpClient.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimPod\PhpSnmp\Transport;
+
+interface SnmpClient
+{
+    /**
+     * @param string[] $oids
+     *
+     * @return array<string, mixed>
+     */
+    public function get(array $oids) : array;
+
+    /**
+     * @param string[] $oids
```

Comment:
```suggestion
     * @param list<string> $oids
```

Reply:
same thing afaik, but as of now not widely supported

Reply:
not same. list has no keys and is supported by all static analysis tooling

Reply:
list has keys, numeric ones... 

not yet supproted by ide. What is the advantage here?

Reply:
the type is more narrow than array. it's the same like typing `object` instead of `mixed`. You narrow the set of possible value types.

jetbrains will support modern php in 2030, no reason to wait for them and write less typed code. There's plugin that does their job https://github.com/Haehnchen/idea-php-generics-plugin

Reply:
ok, sure, the IDE support here isn't that big of a deal